app/src/main/java/com/example/keniphhomemaintenance/dwellings/DwellingsScreenView.kt

```diff
@@ -36,6 +44,14 @@ fun DwellingsScreenView(
             NewDwellingDialog(isDwellingDialogVisible)
         }
     }
+
+    if (!dwellingsViewModel.loadedDwellingsPreviously.value) {
```

Comment:
This is the issue I mentioned regarding the callBack function running after the UI is loaded. If this block here (lines 48-53) are placed directly before the scaffold, then the UI will load with an empty dataset, and once the callback is initiated, the UI will not update.

Reply:
Database should probably be instantiated earlier in the app lifecycle - MainActivity.onCreate() is probably fine for now. Should get rid of the bug.

Reply:
Once we do DI, it won't matter, since the DB should be created by Koin when the app is launched